Packs/GenetecSecurityCenter/Integrations/GenetecSecurityCenterEventCollector/GenetecSecurityCenterEventCollector.py

```diff
@@ -0,0 +1,176 @@
+import demistomock as demisto
+from urllib3 import disable_warnings
+from CommonServerPython import *  # noqa # pylint: disable=unused-wildcard-import
+from CommonServerUserPython import *  # noqa
+from requests.auth import HTTPBasicAuth
+import json
+from xmltodict import parse
+
+disable_warnings()
+
+""" CONSTANTS """
+
+VENDOR = "Genetec"
+PRODUCT = "Security center"
+DATE_FORMAT_EVENT = "%Y-%m-%dT%H:%M:%S"
+AUDIT_TRAIL_ENDPOINT = "/WebSdk/report/AuditTrail"
+GET_EVENTS_HEADERS = [
+    "Id", "Guid", "ModificationTimeStamp", "ModifiedBy", "SourceApplicationGuid", "Name", "ModifiedByAsString",
+    "SourceApplicationAsString", "Machine", "SourceApplicationType", "OldValue", "NewValue", "CustomFieldId", "CustomFieldName",
+    "CustomFieldValueType", "AuditTrailModificationType", "Type", "Description", "AuditTrailModificationSubTypes", "Value"
+]
+
+
+""" CLIENT CLASS """
+
+
+class Client(BaseClient):
+    """Client class to interact with the service API
+
+    :param base_url (str): The server URL.
+    :param username (str): The account username.
+    :param password (str): The account password.
+    :param app_id (str): The app ID.
+    :param max_fetch (str): The maximum number of events to fetch per interval.
+    :param verify (bool): specifies whether to verify the SSL certificate or not.
+    :param proxy (bool): specifies if to use XSOAR proxy settings.
+    """
+
+    def __init__(
+        self, base_url: str, username: str, password: str, verify: bool, proxy: bool, max_fetch: str, app_id: str
+    ):
+        auth = self._encode_authorization(username, password, app_id)
+        self.limit = max_fetch
+
+        super().__init__(base_url=base_url, verify=verify, proxy=proxy, auth=auth, headers={"Accept": "text/json"})
+
+    def _encode_authorization(self, username: str, password: str, app_id: str):
+        updated_username = f"{username};{app_id}"
+        return HTTPBasicAuth(updated_username, password)
+
+    def http_request(self, url_suffix):
+        response_audit = self._http_request('GET', url_suffix=url_suffix, resp_type='response')
+        response = json.loads(response_audit.content)["Rsp"]
+        if response["Status"] == "Fail":
+            raise DemistoException(response["Result"])
+        return response["Result"]
+
+
+""" COMMAND FUNCTIONS """
+
+
+def test_module(client: Client):
+    """
+    Testing we have a valid connection to Genetec security center.
+    """
+    time_now: datetime = datetime.utcnow()
+    start_time: datetime = time_now - timedelta(minutes=1)
+    time_now_str = time_now.strftime(DATE_FORMAT_EVENT)
+    start_time_str = start_time.strftime(DATE_FORMAT_EVENT)
+    query = f"TimeRange.SetTimeRange({start_time_str},{time_now_str})"
+    url_suffix = f"{AUDIT_TRAIL_ENDPOINT}?q={query}"
+    client.http_request(url_suffix=url_suffix)
+    return "ok"
+
+
+def fetch_events_command(
```

Comment:
add logs throughout

Reply:
I have logs for the number of fetched events, the new last_run and the query.. are there any other logs type required?